wit/types.wit

```diff
@@ -1,10 +1,11 @@
-interface messaging-types {
+interface types {
     /// A connection to a message-exchange service (e.g., buffer, broker, etc.).
     resource client {
         connect: static func(name: string) -> result<client, error>;
     }
     
-    /// TODO(danbugs): This should be eventually extracted as an underlying type for other wasi-cloud-core interfaces.
+    // TODO(danbugs): This should be eventually extracted as an underlying type for other 
```

Comment:
I really think we should just have an enum and not a resource for errors here. Allows things to be more descriptive in how they handle errors

Reply:
The idea for an error resource was borrowed from `wasi-keyvalue` at the time. That said, `wasi-keyvalue` is currently using a `variant`, and I think it would be good to follow the same pattern here.